lib/streamlit/DeltaGenerator.py

```diff
@@ -1936,18 +1933,30 @@ def file_uploader(self, element, label, type=['*'], key=None):
 
         Examples
         --------
-        >>> file = st.file_uploader("Upload a image", type=([".png"]))
+        >>> file = st.file_uploader("Upload a image", type=[".png"])
         >>> if file != None:
         >>>     st.image(file)
-
         """
         element.file_uploader.label = label
-        element.file_uploader.type[:] = type
+
+        if type is not None:
+            total_len = sum([len(i) for i in type])
+            if len(type) == total_len: # If all elements are single chars, is a string
+                if type[0] == '.':
+                    element.file_uploader.type[:] = [''.join(type)] # example: type=".csv"
+                else:
+                    element.file_uploader.type[:] = ['.' + ''.join(type)] # example: type="csv"
+            else:
+                element.file_uploader.type[:] = map(lambda x: x if x[0] == '.' else '.' + x , type) # example: type=["png", ".jpeg"]
+
         element.file_uploader.max_upload_size_mb = config.get_option("server.maxUploadSize")
         _set_widget_id("file_uploader", element, user_key=key)
-        progress, data = self._file_manager.get_data(element.file_uploader.id)
-        element.file_uploader.progress = progress
-        return NoValue if data  is None else data
+        ctx = get_report_ctx()
+        if ctx is not None:
+            progress, data = ctx.file_manager.get_data(element.file_uploader.id)
+            element.file_uploader.progress = progress
+
+        return NoValue if data is None else data
```

Comment:
There's a bug here. When `ctx` is `None`, `data` will not be declared and you'll get an exception.